packages/core/lib/frame-context.tsx

```diff
@@ -0,0 +1,51 @@
+import React, {
```

Comment:
I think this file is causing server components to fail on build: https://github.com/puckeditor/puck/actions/runs/16220036922/job/45798124386?pr=1078

We might need to specify use-client.

```suggestion
"use client";

import React, {
```